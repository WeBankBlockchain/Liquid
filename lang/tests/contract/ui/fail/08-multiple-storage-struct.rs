use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop_1 {}

    #[liquid(storage)]
    struct Noop_2 {}

    #[liquid(methods)]
    impl Noop_1 {
        pub fn new(&mut self) {}

        pub fn noop(&self) {}
    }

    #[liquid(methods)]
    impl Noop_2 {
        pub fn new(&mut self) {}

        pub fn noop(&self) {}
    }
}

fn main() {}
