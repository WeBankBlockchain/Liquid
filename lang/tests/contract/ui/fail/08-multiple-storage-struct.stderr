error: duplicate `#[liquid(storage)]` struct definition found here
 --> $DIR/08-multiple-storage-struct.rs:8:5
  |
8 | /     #[liquid(storage)]
9 | |     struct Noop_2 {}
  | |____________________^
